mlflow/utils/search_utils/parse.py

```diff
@@ -0,0 +1,231 @@
+
+    >>> parse_filter_string("params.alpha = '0.1' AND metrics.accuracy > 0.9")
+    [Comparison(KeyType.PARAM, alpha, ComparisonOperator.EQUAL, 0.1),
+     Comparison(KeyType.METRIC, accuracy, ComparisonOperator.GREATER_THAN, 0.9)]
+    """
+
+    try:
+        parsed = sqlparse.parse(string)
+    except Exception:
+        raise MlflowException("Error on parsing filter '{}'".format(string),
+                              error_code=INVALID_PARAMETER_VALUE)
+
+    try:
+        [statement] = parsed
+    except ValueError:
+        raise MlflowException("Invalid filter '{}'. Must be a single statement.".format(string),
+                              error_code=INVALID_PARAMETER_VALUE)
+
+    if not isinstance(statement, SqlStatement):
+        raise MlflowException("Invalid filter '{}'. Must be a single statement.".format(string),
+                              error_code=INVALID_PARAMETER_VALUE)
+
+    return _parse_statement(statement)
+
+
+def _parse_statement(statement):
+    invalids = list(filter(_invalid_statement_token, statement.tokens))
+    if len(invalids) > 0:
+        invalid_clauses = ", ".join("'{}'".format(token) for token in invalids)
+        raise MlflowException("Invalid clause(s) in filter string: {}".format(invalid_clauses),
+                              error_code=INVALID_PARAMETER_VALUE)
+    return [_parse_comparison(token)
+            for token in statement.tokens if isinstance(token, SqlComparison)]
+
+
+def _invalid_statement_token(token):
+    if isinstance(token, SqlComparison):
+        return False
+    elif token.is_whitespace:
+        return False
+    elif token.match(ttype=SqlTokenType.Keyword, values=["AND"]):
+        return False
+    else:
+        return True
+
+
+def _parse_comparison(comparison):
+    stripped_comparison = [token for token in comparison.tokens if not token.is_whitespace]
+
+    try:
+        [identifier, operator, value] = stripped_comparison
+    except ValueError:
+        message = "Invalid comparison clause '{}'. Expected 3 tokens but found {}".format(
+            comparison.value, len(stripped_comparison))
+        raise MlflowException(message, error_code=INVALID_PARAMETER_VALUE)
+
+    try:
+        key_type, key = _parse_identifier(identifier)
+        operator = _parse_operator(operator)
+        value = _parse_value(key_type, value)
+        comparison = Comparison(key_type, key, operator, value)
+    except ValueError as e:
+        raise MlflowException("Invalid comparison clause '{}'. {}".format(comparison.value, e),
+                              error_code=INVALID_PARAMETER_VALUE)
+
+    return comparison
+
+
+def _parse_identifier(token):
+    if not isinstance(token, SqlIdentifier):
+        raise ValueError(INVALID_IDENTIFIER_TPL.format(token=token.value))
+
+    try:
+        key_type_string, key = token.value.split(".", 1)
+    except ValueError:
+        raise MlflowException(INVALID_IDENTIFIER_TPL.format(token=token.value))
```

Comment:
Could we pass `mlflow.protos.databricks_pb2.INVALID_PARAMETER_VALUE` as the second argument to the MlflowException constructor to help clarify the type of error?  Eventually the OSS server may use this field to set a 400-level response code instead of a 500 internal error.